tools/rewriter/SourceRewriter.cpp

```diff
@@ -1165,14 +1256,31 @@ auto ValidateDirectory = CLI::Validator(
     },
     "Directory", "path of a directory");
 
-const std::map<std::string, int> arch_map{{"x86", (int)Arch::X86}, {"aarch64", (int)Arch::Aarch64}};
+const std::map<std::string, Arch> arch_map{{"x86", Arch::X86}, {"aarch64", Arch::Aarch64}};
 
 std::string CompilationDbPath;
+std::string LibraryFilesFile;
+std::string RewriteFilesFile;
 std::vector<std::string> SourceFiles;
 std::vector<std::string> ExtraArgs;
 
+auto LibOnlyGroup = "Library-only mode";
+
 int main(int argc, const char **argv) {
   CLI::App app{"IA2 Source Rewriter"};
+  auto LibFilesOpt =
+      app.add_option("--library-files", LibraryFilesFile, "Path to file listing untrusted library sources")
+          ->group(LibOnlyGroup)
+          ->option_text("FILE")
+          ->check(CLI::ExistingFile);
+  auto RwFilesOpt =
+      app.add_option("--rewrite-files", RewriteFilesFile, "Path to file listing application sources to rewrite")
```

Comment:
Why do you need this to be separate from `source-files`? Could you use that in library-only mode?